tests/test_math.py

```diff
@@ -248,6 +248,14 @@ def test_getitem(mm):
             assert vec[j] == np_mat[j, i]
 
 
+@pytest.mark.parametrize("dtype_", [np.int64, np.double, np.complex128])
+def test_getitem_AnsVec(mm, dtype_):
+    size_i = 3
+    vec = mm.rand(size_i, dtype=dtype_)
+    np_vec = vec.asarray()
+    assert vec[1] == np_vec[1]
```

Comment:
```suggestion
    assert np.allclose(vec, vec.asarray())
```

Reply:
@germa89 looks like `assert np.allclose(vec, vec.asarray()` does not trigger the `__getitem__` and I see codecov issues.
I am reverting back to doing it as per @clatapie suggestions 
`    for i in range(size_i):
        assert vec[i] == np_vec[i]` way so that it triggers the `__get_item__`

Reply:
Hi @kmkoshy! 
I don't think that the codecov issue is coming from the fact that you're using `assert np.allclose(vec, vec.asarray())`. In fact it should be more efficient that the code I suggested.
In my opinion, the problem comes from [here](https://app.codecov.io/gh/pyansys/pymapdl/pull/1882): there is no proper test for a complex vector so that's why the coverage is lower.

Reply:
@clatapie Yes I agree that `assert np.allclose(vec, vec.asarray()` is a better way to do it but since we are not calling the vector by index `vec[i]` this way it does not trigger the `__get_item__` function we updated.
and that is why even thought the test is added to test complex dtype it is showing `there is no proper test for a complex vector `

Reply:
Ow yeah you're right! I looked at that too fast.
Thank you @kmkoshy 😄

Reply:
I'm happy with the change. Well done guys. Thank you @kmkoshy for this nice PR and thank you @clatapie por having a look at it. 🚀